packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts

```diff
@@ -0,0 +1,125 @@
+import { AppError } from '@hicommonwealth/core';
+import { ContestAbi } from './Abi/ContestAbi';
+import ContractBase from './ContractBase';
+import NamespaceFactory from './NamespaceFactory';
+
+class Contest extends ContractBase {
+  namespaceFactoryAddress: string;
+  namespaceFactory: NamespaceFactory;
+
+  constructor(contractAddress: string, factoryAddress: string, rpc: string) {
+    super(contractAddress, ContestAbi, rpc);
+    this.namespaceFactoryAddress = factoryAddress;
+  }
+
+  async initialize(withWallet: boolean = false): Promise<void> {
+    await super.initialize(withWallet);
+    this.namespaceFactory = new NamespaceFactory(
+      this.namespaceFactoryAddress,
+      this.rpc,
+    );
+    await this.namespaceFactory.initialize();
+  }
+
+  /**
+   * deploys a new recurring contest and configures the feeManager to divert feeShare of
+   * stake fees to new contest
+   * @param namespaceName namespace name to configure for
+   * @param contestInterval the recurrng contest interval in ms
+   * @param winnerShares the percent of each respective winning places prize(ie 5 = 5%)
+   * @param stakeId the id of the community stake token, defaults to 2
+   * @param prizeShare the % amount of the prize pool to payout each interval(ie 5 = 5%)
+   * @param voterShare the % amount of prize pool claimable by voters each week(ie 5 = 5%)
+   * @param feeShare the % amount of community stake fees to divert to contest(ie 5 = 5%)
+   * @param weight the weight of each stake balance, defaults to 1
+   * @param walletAddress the wallet address to make transactions from
+   * @returns the contract address of the new contest
+   */
+  async newRecurringContest(
+    namespaceName: string,
+    contestInterval: number,
+    winnerShares: number[],
+    stakeId: number = 2,
+    prizeShare: number,
+    voterShare: number,
+    feeShare: number,
+    weight: number = 1,
+    walletAddress: string,
+  ): Promise<string> {
+    if (!this.initialized || !this.walletEnabled) {
+      await this.initialize(true);
+    }
+    try {
+      const txReceipt = await this.namespaceFactory.newContest(
+        namespaceName,
+        contestInterval,
+        winnerShares,
+        stakeId,
+        prizeShare,
+        voterShare,
+        feeShare,
+        weight,
+        walletAddress,
+      );
+      const newContestAddress =
+        txReceipt.events.NewContest.returnValues.contest;
+      this.contractAddress = newContestAddress;
+      return newContestAddress;
+    } catch (error) {
+      throw new Error('Failed to initialize contest ' + error);
+    }
+  }
+
+  //TODO Reccurring contest pending ABI
```

Comment:
are we expecting more code here?

Reply:
Ah weird. I think one of my commits was pushed. Will reconcile before moving from draft